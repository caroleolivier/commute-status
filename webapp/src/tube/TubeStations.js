import React, { Component } from 'react';
import PropTypes from 'prop-types';

import TubeStationContainer from './TubeStationContainer';

class TubeStations extends Component {
    render() {
        const stations = this.props.tubeStations.map(
            station => <TubeStationContainer key={station.stationId} tubeStation={station} />
        );
        return (
            <div>
                <h3>Tube(s)</h3>
                {stations}
            </div>
        );
    }
}

TubeStations.propTypes = {
    tubeStations: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default TubeStations;
