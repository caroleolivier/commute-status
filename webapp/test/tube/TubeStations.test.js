/* eslint-env browser, jest */

import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import TubeStations from '../../src/tube/TubeStations';

describe('TubeStations component', () => {
    test('shallow render matches snapshot.', () => {
        const tubeStations = [
            { stationName: 'Houston', stationId: '12345', direction: 'inbound', lineName: 'northern' }
        ];
        const busStationContainer = shallow(<TubeStations tubeStations={tubeStations} />);
        const tree = toJson(busStationContainer);
        expect(tree).toMatchSnapshot('rendering');
    });
});
