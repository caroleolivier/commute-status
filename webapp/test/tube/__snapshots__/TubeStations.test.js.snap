// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering 1`] = `
<div>
  <h3>
    Tube(s)
  </h3>
  <TubeStationContainer
    key="12345"
    tubeStation={
      Object {
        "direction": "inbound",
        "lineName": "northern",
        "stationId": "12345",
        "stationName": "Houston",
      }
    }
  />
</div>
`;
